25.js

```diff
@@ -0,0 +1,157 @@
+// Create class UserService
+// Implement every scenario from dummy users BE in your class
+// For each scenario create new method
+// Methods can be static
+// Read response from BE and console.log it or store it in class properties
+// Don't forget to catch errors when request failed
+
+// *Extra task: Delete users with ids 1,2,3,4 and random id from 5 to 10000 simultaneously
+// Use Promise.allSettled for this task
+
+class UserService {
+  users = [];
+  selectedUser = null;
+  searchedUsers = [];
+  filteredUsers = [];
+  carts = null;
+  posts = null;
+  todos = null;
+  page = 0;
+
+  getAllUsers() {
+    fetch("https://dummyjson.com/users", {
+      method: "GET",
+    })
+      .then((res) => {
+        console.log(`${res.status} ${res.url}`);
+        return res.json();
+      })
+      .then((data) => {
+        this.users.push(...data.users);
+      });
+  }
+  getUser(id) {
+    fetch(`https://dummyjson.com/users/${id}`)
+      .then((res) => {
+        console.log(`${res.status} ${res.url}`);
+        return res.json();
+      })
+      .then((data) => {
+        this.selectedUser = data;
+      });
+  }
+  searchUser(param) {
+    const url = new URL("https://dummyjson.com/users/search");
+    url.searchParams.set("q", param);
+    fetch(url)
+      .then((res) => {
+        console.log(`${res.status} ${res.url}`);
+        return res.json();
+      })
+      .then((data) => {
+        this.searchedUsers.push(...data.users);
+      });
+  }
+  filterUser(param) {
+    const url = new URL("https://dummyjson.com/users/filter");
+    url.searchParams.set("key", param.key);
+    url.searchParams.set("value", param.value);
+    fetch(url)
+      .then((res) => {
+        console.log(`${res.status} ${res.url}`);
+        return res.json();
+      })
+      .then((data) => {
+        this.filteredUsers.push(...data.users);
+      });
+  }
+  async paginatedUsers(limit) {
```

Comment:
it's ok to practice but in future choose one way of handling promises. 
It's better to have one code style and don't mix styles in one class, function or file
I mixed async handling methods only to show both ways